pkg/istio-agent/agent.go

```diff
@@ -228,9 +240,91 @@ func (a *Agent) initializeEnvoyAgent() error {
 
 	drainDuration, _ := types.DurationFromProto(a.proxyConfig.TerminationDrainDuration)
 	a.envoyAgent = envoy.NewAgent(envoyProxy, drainDuration)
+	a.envoyWaitCh = make(chan error, 1)
+	if a.cfg.EnableDynamicBootstrap {
+		// Simulate an xDS request for a bootstrap
+		go func() {
+			// wait indefinitely and keep retrying with jittered exponential backoff
+			backoff := 500
+			max := 30000
+			request := &bootstrapDiscoveryRequest{
+				node:        node,
+				envoyWaitCh: a.envoyWaitCh,
+				envoyUpdate: envoyProxy.UpdateConfig,
+			}
+			for {
+				_ = a.xdsProxy.handleStream(request)
+				if request.received {
+					break
+				}
+				request.sent = false
+				delay := time.Duration(rand.Int()%backoff) * time.Millisecond
```

Comment:
the jitter here seems huge - backoff is going from (for example) 27s, 9s, 18s, 3ms.

Can we just use github.com/cenkalti/backoff for consistency with the rest of the code? It has a jitter but its a smaller one, just jittering like 10% off a perfect exponential backoff

Reply:
Sure, but it's such a trivial function that a dependency is an overkill. FYI, there's a lot of variation in jittering:
- AWS recommends this version of jitter: uniform random [0, backoff]
- GCP jitters slightly up: backoff + uniform random [0, 1s]
- gRPC lib jitters up and down by 20%: backoff + uniform random [ -.2 backoff, .2 backoff]

I'll change it to be like gRPC.